tests/js/articles.frontend.spec.js

```diff
@@ -0,0 +1,139 @@
+import {
+  createNewPost,
+  enablePageDialogAccept,
+  setPostContent,
+  publishPost,
+} from '@wordpress/e2e-test-utils';
+
+import {
+  selectBlockByName,
+  openSidebarPanelWithTitle,
+  clickElementByText,
+  typeInDropdownWithLabel,
+} from './e2e-tests-helpers';
+
+import 'expect-puppeteer';
+
+const ARTICLS_BLOCK = `
+<!-- wp:planet4-blocks/articles {"article_count":2} -->
+<div class="wp-block-planet4-blocks-articles" data-render="planet4-blocks/articles" data-attributes="{&quot;attributes&quot;:{&quot;article_heading&quot;:&quot;Related Articles&quot;,&quot;article_count&quot;:2,&quot;tags&quot;:[],&quot;posts&quot;:[],&quot;post_types&quot;:[],&quot;read_more_text&quot;:&quot;Load More&quot;,&quot;read_more_link&quot;:&quot;&quot;,&quot;button_link_new_tab&quot;:false,&quot;ignore_categories&quot;:false},&quot;innerBlocks&quot;:[]}"></div>
+<!-- /wp:planet4-blocks/articles -->
+`;
+
+describe( 'Articles block frontend', () => {
+  beforeAll( async () => {
+    // This helps in overriding Wordpress dialogs preventing actions.
+    await enablePageDialogAccept();
+  } );
+
+  it ( 'test post thumbnail behaviour', async () => {
+    // Create new post.
+    await createNewPost({
+      title: 'Test Articles with no thumbnail',
+      content: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+      showWelcomeGuide: false,
+    });
+    await publishPost();
+    await page.waitForSelector( '.post-publish-panel__postpublish' );
+
+    //Create a new page and add newly created post in it.
+    await createNewPost({
+      postType: 'page',
+      title: 'Test Articles block thumbnail',
+      showWelcomeGuide: false,
+    });
+
+    // Add articles block in page.
+    await setPostContent( ARTICLS_BLOCK );
```

Comment:
Would adding the post content in `createNewPost` directly, on previous line, have the same result?

Reply:
With `createNewPost` the blocks comments are treated as text data, I tried that earlier.